pkg/storage/chunk/client/local/boltdb_index_client.go

```diff
@@ -258,69 +258,103 @@ func (b *BoltIndexClient) QueryDB(ctx context.Context, db *bbolt.DB, bucketName
 }
 
 func (b *BoltIndexClient) QueryWithCursor(_ context.Context, c *bbolt.Cursor, query index.Query, callback index.QueryPagesCallback) error {
-	var start []byte
-	if len(query.RangeValuePrefix) > 0 {
-		start = []byte(query.HashValue + separator + string(query.RangeValuePrefix))
-	} else if len(query.RangeValueStart) > 0 {
-		start = []byte(query.HashValue + separator + string(query.RangeValueStart))
-	} else {
-		start = []byte(query.HashValue + separator)
-	}
+	batch := batchPool.Get().(*cursorBatch)
+	batch.reset()
+	batch.cursor = c
+	batch.query = &query
```

Comment:
Should we pass these to `batch.reset`? Would make it a bit clearer I feel.

Reply:
yep